src/App.js

```diff
@@ -3,18 +3,84 @@ import Card from './Card';
 import { useState } from 'react';
 
 function App() {
-  const [checked, setChecked] = useState(false);
+  const [cardState, setCardState] = useState({
+    checked: false,
+    isEditable: false
+  });
+
+  const [cardText, setCardText] = useState({
+    headerText: 'Caption',
+    tempHeader: 'Caption',
+    bodyText: 'text...',
+    tempBody: 'text...'
+  });
+
   const changeStyleHandler = () => {
-    setChecked(!checked);
-  }
+    setCardState({
+      ...cardState,
+      checked: !cardState.checked,
+    });
+  };
+
+  const openEditModeHandler = () => {
+    setCardState({
+      checked: false,
+      isEditable: !cardState.isEditable
+    });
+  };
+
+  const headerTextChangeHandler = e => {
+    setCardText({
+      ...cardText,
+      headerText: e.target.value
+    });
+  };
+
+  const bodyTextChangeHandler = e => {
+    setCardText({
+      ...cardText,
+      bodyText: e.target.value
+    });
+  };
+
+  const saveChangesHandler = () => {
+    setCardState({
+      ...cardState,
+      isEditable: !cardState.isEditable
+    });
+    setCardText({
+      ...cardText,
+      tempHeader: cardText.headerText,
+      tempBody: cardText.bodyText
```

Comment:
Like here, if you separate `tempData `and `cardData `objects are make properties the same, you can just use a spread operator to apply changes. Not so significant here, just imagine if there are couple dozens fields.